### PR TITLE
Feature/homology annotation Incorporate datachecks into reference update pipeline

### PR DESCRIPTION
**Pull request by CristiGuijarro**
## Description

In the Rapid Release the datachecks are not going to be run manually on each per species compara database so the datachecks need to be within the pipeline.

**Related JIRA tickets:**
- ENSCOMPARASW-4343

## Overview of changes
Parameter and tweaks in `UpdateReferenceDatabase_conf`: new results table, DC specific parameters, tweaks_analyses sub to manipulate runnable dataflows and parameters from the DC Parts/. Turn off `jira_ticket_creation` and pass specific parameters to specific runnables only.

## Testing
In[ this pipeline ](http://guihive.ebi.ac.uk:8080/versions/96/?driver=mysql&username=ensadmin&host=mysql-ens-compara-prod-10&port=4648&dbname=cristig_update_references_dcs&passwd=xxxxx)using `mysql-ens-compara-prod-3 jalvarez_reference_db_test_rename_2` as test ref_db.

## Notes
Not all the dcs passed (which was expected and a little intentional), these were forgiven so that the whole pipeline could complete.
I am initially basing this against `feature/homology_annotation_ja` because the changes are rebased onto changes made in this branch. I can swap this back over to `feature/homology_annotation` once @JAlvarezJarreta PR is submitted and approved or we could consider approving and merging this into `feature/homology_annotation_ja`.

### CODE REVIEW

File: modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm
Comment (JAlvarezJarreta) [inline modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm:284]: We need to make sure this line disappears before merging (I guess it is here because of the conflicts we have talked about before).

File: modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm
Comment (JAlvarezJarreta) [inline modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm:359]:
```suggestion
    $analyses_by_name->{'store_results'}->{'-parameters'}->{'dbname'} = '#ref_db#';
```

**Review by JAlvarezJarreta — COMMENTED** [review]
Now adding DCs seems really easy! The only question I have after reviewing these two PRs is if the rewiring should be done the other way around: by default the DC code should be "unwired" and then you wire it for the DC pipeline (only). What do you think?

File: modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm
Comment (CristiGuijarro) [reply modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm:359]: The `dbname` has to explicitly be the `dbname` only - specific to the production side runnable - I hard-coded it here because I figure the name will not change for the reference db- but happy to include it as a parameter - to note the `dbname` cannot be seen elsewhere or it will need to be explicitly passed as `undef` to the other runnables to work :grimacing: 
Alternatively I could put a wrapper around the production runnable to collect the dbname - but this is only necessary to this specific pipeline because the dcs are running on the ref_db and not the pipelin_db.

**Comment by CristiGuijarro** [comment]
> Now adding DCs seems really easy! The only question I have after reviewing these two PRs is if the rewiring should be done the other way around: by default the DC code should be "unwired" and then you wire it for the DC pipeline (only). What do you think?

I was considering this even with PR #261 - but there would still need to be rewiring done specific to each pipeline. This rewiring would not be necessary for the smaller DC parts - which we would be able to integrate into pipelines between analyses - where it would be more useful. So I am sitting nicely balanced on the fence.

**Review by JAlvarezJarreta — APPROVED** [review]
Pre-approved, pending the decision about those two comments.

**Comment by JAlvarezJarreta** [comment]
Commit 45c7609 in your commit history shouldn't be there though.

File: modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm
Comment (CristiGuijarro) [reply modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm:97]: I think keep it here, because if we decide later on to do the re-wiring mentioned, it will be easier to keep track and change if they are all together?

File: modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm
Comment (CristiGuijarro) [inline modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateReferenceDatabase_conf.pm:359]:
```suggestion
    $analyses_by_name->{'store_results'}->{'-parameters'}->{'dbname'} = $self->o('ref_dbname');
```

**Pull request merged**
